critiquebrainz/db/oauth_client.py

```diff
@@ -0,0 +1,137 @@
+from critiquebrainz import db
+from critiquebrainz.db import exceptions as db_exceptions
+from critiquebrainz.utils import generate_string
+import sqlalchemy
+
+
+def create(*, user_id, name, desc, website, redirect_uri):
+    """Create new OAuth client and generates secret key for it.
+
+    Args:
+        user_id (uuid): ID of the user who manages the client.
+        name (str): Name of the client.
+        desc (str): Client description.
+        website (str): Client web site.
+        redirect_uri: URI where responses will be sent.
+    Returns:
+        Dictionary of the following structure
+        {
+            "client_id": str,
+            "client_secret": str,
+            "user_id": uuid,
+            "name": str,
+            "desc": desc,
+            "website": website,
+            "redirect_uri": redirect_uri,
+        }
+    """
+    client_id = generate_string(20)
+    client_secret = generate_string(40)
```

Comment:
These variables seem unnecessary. Calls to `generate_string` can be moved into the dictionary below.

I'd also move lengths into "constants" in this module: `CLIENT_ID_LENGTH` and `CLIENT_SECRET_LENGTH`.